corehq/messaging/scheduling/views.py

```diff
@@ -871,6 +874,27 @@ def post(self, request, *args, **kwargs):
                     "does not have access to inbound SMS"
                 )
 
+            # list of fields that are getting changed everytime a request came manually from django form.
+            changed_schedule_fields = [
+                'send_time_type', 'repeat', 'stop_type', 'use_user_data_filter',
+                'start_offset_type', 'start_day_of_week', 'visit_window_position',
+                'capture_custom_metadata_item', 'stop_date_case_property_enabled'
+            ]
+
+            user_changed_data = self.schedule_form.changed_data
+            edit_mode = kwargs.get('edit_mode')
+            if (
+                ((user_changed_data != changed_schedule_fields) or self.criteria_form.has_changed())
+                and edit_mode and not submit_form
+            ):
+
+                case_type = self.criteria_form.cleaned_data.get('case_type')
+                case_count = CaseES().domain(self.domain).case_type(case_type).count()
+
+                warning_msg = _(f'This update may alert {case_count} cases. Are you sure you want to proceed?')
+                kwargs.update({'warning_msg': warning_msg})
+                return self.get(request, *args, **kwargs)
+
```

Comment:
is this working as expected? This logic to populate the `warning_msg` is in "post" which is after the request has been submitted. Isn't the requirement to show user the message when they are submitting the request and hence it should be added when the edit view is being loaded, no?

also, we are still doing the `changed_schedule_fields`?

Reply:
i am guessing all the logic to populate the message would go in `page_context` in `EditConditionalAlertView` and you wont need the `edit_mode` param.

Reply:
Agreed with this.

Also, the `confirmationFlag` logic also doesn't belong server-side: that suggests there's a scenario where the page reloads but the form doesn't actually do anything. You should be able to handle the confirmation in javascript, likely by using a `submit` handler on the form and returning false if the user clicked "No" in the popup (that'll prevent the form from being submitted).

Reply:
Yes it is working as expected. I think we cannot put this logic inside edit views because edit view is loaded at the moment user clicks on conditional alert name to edit it but we need to check the updated value which we will get once the user clicks on save button.

even if we are putting the code inside page_context in edit view the same problem will be there as explained above.


I am verifying the updated value in server side because I am not able do that at javascript side, I mean I did not get any idea to verify all the updated fields except name and message (we dont have to show pop if these two are updated) in javascript.

The confirmationFlag is helping me to identify that the user has clicked on yes button on popup and I have to let them submit the form successfully. if confirmationFlag is coming as blank then with the help of some other flags like edit_mode I am deciding whether I should show popup (in case of some fields updated) or should not show the popup (in case of new conditional alert is created).